dbt_platform_helper/custom_resources/app_user.py

```diff
@@ -0,0 +1,204 @@
+                Value=json.dumps(user_secret_string),
+                Overwrite=True,
+            )
+
+    return user_secret
+
+
+def send(event, body, logger, headers):
+    send = request.Request(event["ResponseURL"], data=body, headers=headers)
+    send.get_method = lambda: "PUT"
+
+    count = 0
+    while count < 5:
+        count += 1
+        try:
+            request.urlopen(send)
+            break
+        except HTTPError as ex:
+            if count < 5:
+                logger.warning(f"{ex} [{ex.url}] - Retry {count}")
+            else:
+                logger.error(f"{ex} [{ex.url}]")
+            time.sleep(count * 5)
+
+
+# borrowed from https://github.com/awslabs/aws-cloudformation-templates/blob/master/aws/services/CloudFormation/MacrosExamples/StackMetrics/lambda/cfnresponse.py
+# tweaked to use requests library
+def send_response(
+    event, context, responseStatus, responseData, physicalResourceId=None, noEcho=False, reason=None
+):
+    responseUrl = event["ResponseURL"]
+
+    print(responseUrl)
+
+    responseBody = {}
+    responseBody["Status"] = responseStatus
+    responseBody["Reason"] = (
+        reason or "See the details in CloudWatch Log Stream: " + context.log_stream_name
+    )
+    responseBody["PhysicalResourceId"] = physicalResourceId or context.log_stream_name
+    responseBody["StackId"] = event["StackId"]
+    responseBody["RequestId"] = event["RequestId"]
+    responseBody["LogicalResourceId"] = event["LogicalResourceId"]
+    responseBody["NoEcho"] = noEcho
+    responseBody["Data"] = responseData
+
+    json_responseBody = json.dumps(responseBody)
+
+    print("Response body:\n" + json_responseBody)
+
+    headers = {"content-type": "", "content-length": str(len(json_responseBody))}
+
+    send(event, json_responseBody.encode(), logger, headers)
+
+
+def handler(event, context):
+    print("REQUEST RECEIVED:\n" + json.dumps(event))
+
+    db_master_user_secret = event["ResourceProperties"]["MasterUserSecret"]
+    user_secret_name = event["ResourceProperties"]["SecretName"]
+    username = event["ResourceProperties"]["Username"]
+    user_permissions = event["ResourceProperties"]["Permissions"]
+
+    secrets_manager = boto3.client("secretsmanager")
+    ssm = boto3.client("ssm")
+
+    master_user = json.loads(
+        secrets_manager.get_secret_value(SecretId=db_master_user_secret)["SecretString"]
+    )
+
+    user_password = secrets_manager.get_random_password(
+        PasswordLength=16,
+        ExcludeCharacters='[]{}()"@/\;=?&`><:|#',
+        ExcludePunctuation=True,
+        IncludeSpace=False,
+    )["RandomPassword"]
+
+    user_secret_string = {
+        "username": username,
+        "password": user_password,
+        "engine": master_user["engine"],
+        "port": master_user["port"],
+        "dbname": master_user["dbname"],
+        "host": master_user["host"],
+    }
+
+    if "dbClusterIdentifier" in master_user.keys():
```

Comment:
These if blocks are new because the svc addon is shared between the two types of postgres addon